p2p/src/network/kad/stream/p2p_network_kad_stream_reducer.rs

```diff
@@ -31,30 +31,32 @@ impl P2pNetworkKadIncomingStreamState {
             }
             (S::WaitingForRequest { .. }, A::IncomingData { data, .. }) => {
                 let data = &data.0;
-
                 let mut reader = BytesReader::from_bytes(data);
-                let Ok(len) = reader.read_varint32(data).map(|v| v as usize) else {
+
+                let Ok(encoded_len) = reader.read_varint32(data).map(|v| v as usize) else {
                     *self = S::Error(P2pNetworkStreamProtobufError::MessageLength);
                     return Ok(());
                 };
 
-                if len > limits.kademlia_request() {
+                let remaining_len = reader.len();
+
+                if encoded_len > limits.kademlia_request() {
                     *self = S::Error(P2pNetworkStreamProtobufError::Limit(
-                        len,
+                        encoded_len,
                         limits.kademlia_request(),
                     ));
                     return Ok(());
                 }
 
-                if len > reader.len() {
+                if encoded_len > remaining_len {
                     *self = S::PartialRequestReceived {
-                        len,
-                        data: data[(len - reader.len())..].to_vec(),
+                        len: encoded_len,
+                        data: data[(data.len() - remaining_len)..].to_vec(),
                     };
                     return Ok(());
                 }
 
-                self.handle_incoming_request(len, &data[data.len() - reader.len()..])
+                self.handle_incoming_request(encoded_len, &data[data.len() - remaining_len..])
```

Comment:
```suggestion
                self.handle_incoming_request(encoded_len, data.get(data.len() - remaining_len..).unwrap()) // Without the `unwrap`
```

Is it parsing incoming data ? We should probably use `slice::get` instead ?
On line 54 too

Reply:
It seems to be decoding the length from incoming data and then using the rest of it. Maybe I could just use the reader with `.read_message_by_len()`

Reply:
Modified to use `slice::get` now. I think the new else condition is unreachable but I don't think that will be an issue.